app/uk/gov/hmrc/uploaddocuments/controllers/BaseController.scala

```diff
@@ -59,21 +59,37 @@ abstract class BaseController(
 
   final protected def controllerComponents: MessagesControllerComponents = components.messagesControllerComponents
 
-  final def journeyIdFromSession(implicit rh: RequestHeader): Option[JourneyId] =
-    decodeHeaderCarrier(rh).sessionId.map(_.value).map(SHA256.compute)
+  private val journeyIdPathParamRegex = ".*?/journey/([a-fA-F0-9]+?)/.*".r
+
+  final def journeyId(implicit rh: RequestHeader): Option[String] =
+    journeyId(decodeHeaderCarrier(rh), rh)
+
+  private def journeyIdFromPath(implicit rh: RequestHeader) =
+    rh.path match {
+      case journeyIdPathParamRegex(id) => Some(id)
+      case _                           => None
+    }
+
+  private def journeyId(hc: HeaderCarrier, rh: RequestHeader): Option[String] =
+    journeyIdFromPath(rh).orElse(hc.sessionId.map(_.value).map(SHA256.compute))
 
   final implicit def context(implicit rh: RequestHeader): HeaderCarrier = {
     val hc = decodeHeaderCarrier(rh)
-    journeyIdFromSession.fold(hc)(jid => hc.withExtraHeaders("FileUploadJourney" -> jid))
+    journeyId(hc, rh)
+      .map(jid => hc.withExtraHeaders("FileUploadJourney" -> jid))
+      .getOrElse(hc)
```

Comment:
is this the actual change that's necessary?

Reply:
I've just confirmed, this is the only change that was necessary, I'm going to push a commit to undo the revert to `whenInSession`

Reply:
This is used any method that needs an implicit HeaderCarrier, e.g `whenAuthenticated` and the connectors. Once we sort out the implicit JourneyId perhaps we can get rid of the regex again, by passing the implicitly jouneyId into the method. The only problem is this is an implicit def so, as we've found, debugging it can be problematic.